### ISSUE TITLE
[PAY 3RD APRIL] [$250] Upload Workspace Avatar Error is impossible to dismiss

### ISSUE THREAD

**Issue by grgia** [Expensify/App#37819]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

<img width="186" alt="image" src="https://github.com/Expensify/App/assets/38015950/9c00fe5c-de5f-4e7b-abe9-6509f00707ce">

https://github.com/Expensify/App/assets/38015950/3185e7a1-04a5-471d-b1d1-cfc57df280ff

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

## Action Performed:
Break down in numbered steps

Go to workspaces and click on one of your workspaces to open the workspace profile page. 
Force an error in the upload process (I have dev settings that are borked, which caused this error)

## Expected Result:
Describe what you think should've happened

Error should be dismissable/readable

## Actual Result:
Describe what actually happened

Error overlaps the other info, error is not dismissible due to highlightable menu items

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

Workspace uploading works well on staging/prod so this isn't seen much. But we should fix the UI here.

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0185f43e02fc3ad4e8</li>
        <li>Upwork Job ID: 1765394695917617152</li>
        <li>Last Price Increase: 2024-03-06</li>
    </ul>
</details>

**Comment by grgia** [comment]
@shawnborton could you confirm what the error UI pattern should be on this page?

**Comment by grgia** [comment]
This will be external, but we need to hold on design confirmation.

**Comment by shawnborton** [comment]
I would think this would just sit cleanly as it's own full-width row below the avatar. Basically taking the same width as the title row below it. 

cc @narefyev91 - given that you worked on this page, is this something you could fix quickly?

**Comment by godofoutcasts94** [comment]
Hey @grgia! As mentioned in the [comment](https://github.com/Expensify/App/issues/37819#issue-2171447038). Can you please provide the dev setting that gave you that error ?

**Comment by grgia** [comment]
@godofoutcasts94 you can hard code an error in this block 

https://github.com/Expensify/App/blob/a04481ffef2c60243c8c5680222a01a82e162916/src/pages/workspace/WorkspaceProfilePage.tsx#L120-L128

**Comment by godofoutcasts94** [comment]
> @godofoutcasts94 you can hard code an error in this block
> 
> https://github.com/Expensify/App/blob/a04481ffef2c60243c8c5680222a01a82e162916/src/pages/workspace/WorkspaceProfilePage.tsx#L120-L128

Thanks I will do it

**Comment by ShridharGoel** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Upload Workspace Avatar Error is impossible to dismiss.

### What is the root cause of that problem?
Error is being set here in WorkspaceProfilePage:
`errors={policy?.errorFields?.avatar}`

It is then being passed to
```
const setError = (error: TranslationPaths | null, phraseParam: Record<string, unknown>) => {
        setErrorData({
            validationError: error,
            phraseParam,
        });
    };
```

### What changes do you think we should make in order to solve the problem?
Add a close icon next to the error after discussion with design team.
Clicking on the icon would call this:
```
setError(null, {});
```
This will dismiss the message.

**Comment by grgia** [comment]
@jjcoffee 

cc @dukenv0307 @Krishna2323 @ShridharGoel 

Something to keep in mind: the final design will need to align with this description:

> I would think this would just sit cleanly as it's own full-width row below the avatar. Basically taking the same width as the title row below it.

Particularly this part: **taking the same width as the title row below it.**

**Comment by jjcoffee** [comment]
@shawnborton If you have time could you take a quick look at the couple of proposals above? I'm wondering if you meant that the `X` icon to close should sit at the very end of the row or not.

**Comment by shawnborton** [comment]
Basically this mock looks good to me:
![image](https://github.com/Expensify/App/assets/2319350/4c7db65f-8c88-4a01-afac-4fa2dc9aaaa1)

Though I might add just a little more margin/padding above the error message. Like 4px more.

**Comment by shawnborton** [comment]
And maybe a little less margin below it.

**Comment by grgia** [comment]
Thanks for the speedy update @dukenv0307, all yours!

**Comment by grgia** [comment]
hmmm @jliexpensify not sure why the offer automation failed here

**Comment by jliexpensify** [comment]
All good, have manually hired!

**Comment by jjcoffee** [comment]
@grgia Was there any reason why the issue price was set to $250? This required quite a bit of back and forth on the PR to get right, so I think the standard $500 might be more appropriate here. :bow:

**Comment by jliexpensify** [comment]
It looks like this is still ongoing, but I'm OOO 21st to 31st. I'm going to keep this assigned to myself - I suspect that by the time I'm back, this will need payment.

**Comment by jjcoffee** [comment]
[Deployed](https://github.com/Expensify/App/pull/37874#issuecomment-2021926883) to production.

**Comment by jliexpensify** [comment]
@grgia - Hi, any thoughts on @jjcoffee's comment [here](https://github.com/Expensify/App/issues/37819#issuecomment-1991418687)? 

@jjcoffee - does a checklist need to be completed?

**Comment by jjcoffee** [comment]
- The PR that introduced the bug has been identified. Link to the PR: Looks like this behaviour was always lurking
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **N/A**
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/A**
- Determine if we should create a regression test for this bug. **Yes**
- If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**Regression Test Proposal**

1. Go to Workspaces and tap any workspace to open the workspace profile page.
2. Force an error by running the following in the console, replacing `POLICYID` with the workspace's policy ID:
```
Onyx.merge('policy_POLICYID', {
    errorFields: {
        avatar: {
            12345679: "Test error message",
        }
    }
})
```
3. Verify that the error displays filling all the available space, to match the option rows below.

Do we agree 👍 or 👎

**Comment by grgia** [comment]
@jjcoffee, the price reflects the priority of this bug. I do appreciate the work you put into PR reviews; it does not go unnoticed! That said, I do think $250 is fair for this PR, given this bug is low priority, mostly minor styling tweaks, and rarely experienced by users. Does that sound alright to you?

**Comment by jjcoffee** [comment]
@grgia Understood, many thanks for the explanation!

**Comment by jliexpensify** [comment]
Payment Summary:

- Contributor: @dukenv0307 $250 (can you accept 1 of the offers?)
- C+: @jjcoffee $250

[UPWORKS JOB](https://www.upwork.com/jobs/~0185f43e02fc3ad4e8)

**Comment by jliexpensify** [comment]
Ok, I need to make a new Upworks Job because the old one's closed. Resending invites to @dukenv0307 and @jjcoffee

**Comment by jliexpensify** [comment]
Payment Summary:

Contributor: @dukenv0307 $250 
C+: @jjcoffee $250

[New Upworks job](https://www.upwork.com/jobs/~01fbaf51c0273dd53b)

**Comment by jjcoffee** [comment]
@jliexpensify Accepted, thanks!

**Comment by jliexpensify** [comment]
Cheers, just waiting on @dukenv0307

**Comment by jliexpensify** [comment]
@dukenv0307 can you accept the new offer so I can get this closed off? Cheers.

**Comment by jliexpensify** [comment]
Paid and job closed
